Explorer/Assets/DCL/Ipfs/IpfsRealm.cs

```diff
@@ -0,0 +1,127 @@
+                ContentBaseUrl = URLBuilder.Combine(ContentBaseUrl, URLSubdirectory.FromString("contents/"));
+            }
+            else
+            {
+                ContentBaseUrl = URLBuilder.Combine(CatalystBaseUrl, URLSubdirectory.FromString("content/contents/"));
+                EntitiesActiveEndpoint = URLBuilder.Combine(CatalystBaseUrl, URLSubdirectory.FromString("content/entities/active"));
+            }
+        }
+
+        public bool Equals(IpfsRealm other)
+        {
+            if (ReferenceEquals(null, other)) return false;
+            if (ReferenceEquals(this, other)) return true;
+            return CatalystBaseUrl == other.CatalystBaseUrl;
+        }
+
+        public override bool Equals(object obj)
+        {
+            if (ReferenceEquals(null, obj)) return false;
+            if (ReferenceEquals(this, obj)) return true;
+            if (obj.GetType() != GetType()) return false;
+            return Equals((IpfsRealm)obj);
+        }
+
+        public override int GetHashCode() =>
+            ContentBaseUrl.GetHashCode();
+
+        public async UniTask PublishAsync<T>(IpfsRealmEntity<T> entity, IReadOnlyDictionary<string, byte[]> contentFiles, CancellationToken ct)
+        {
+            string entityJson = JsonUtility.ToJson(entity);
+            byte[] entityFile = Encoding.UTF8.GetBytes(entityJson);
+            string entityId = entityFile.IpfsHashV1();
+            AuthChain authChain = web3IdentityCache.Identity!.Sign(entityId);
+
+            var form = new WWWForm();
+
+            form.AddField("entityId", entityId);
+
+            var i = 0;
+
+            foreach (AuthLink link in authChain)
+            {
+                form.AddField($"authChain[{i}][type]", link.type.ToString());
+                form.AddField($"authChain[{i}][payload]", link.payload);
+                form.AddField($"authChain[{i}][signature]", link.signature ?? "");
+                i++;
+            }
+
+            files.Clear();
+
+            foreach ((string key, byte[] value) in contentFiles)
+                files[key] = value;
+
+            files[entityId] = entityFile;
+
+            foreach ((string hash, byte[] data) in files)
+                form.AddBinaryData(hash, data);
+
+            urlBuilder.Clear();
+
+            URLAddress url = urlBuilder.AppendDomain(CatalystBaseUrl)
+                                       .AppendPath(URLPath.FromString("content/entities"))
+                                       .Build();
+
+            try
+            {
+                await webRequestController.PostAsync(new CommonArguments(url),
+                    GenericPostArguments.CreateWWWForm(form), ct);
+            }
+            finally
+            {
+                urlBuilder.Clear();
+                files.Clear();
+            }
+        }
+
+        public async UniTask<string> GetFileHashAsync(byte[] file, CancellationToken ct) =>
```

Comment:
I agree to be sync. Regarding to hashing, I could use the extension method directly: `file.IpfsHashV1()`. I preferred to encapsulate the hashing into the realm implementation, in case it needs to change, thus not tie it into a specific class. I prefer to keep it an instance method. What do you think?

Reply:
alright